pkg/plugininventory/sqlite_inventory.go

```diff
@@ -692,6 +697,16 @@ func (b *SQLiteInventory) InsertPluginGroup(pg *PluginGroup, override bool) erro
 				}
 			}
 
+			// Activate plugins on plugin group publish if the TANZU_CLI_ACTIVATE_PLUGINS_ON_PLUGIN_GROUP_PUBLISH
+			// environment variable is set to True.
+			// Note: If it is set to false, plugins won't be deactivated
+			if activatePlugins {
+				err = b.updatePluginActivationState(db, pi.Name, string(pi.Target), pi.Version, true)
```

Comment:
But in this case, we do want the pipeline to fail, no !?

If the user chooses `skipPGVerification=true` it means plugins are likely not present in the repository and in that case we do not want users to partially activate plugins that are available in staging and leave some of the plugins in production as deactivated. If we allow it, it might give the wrong impression to the users that all plugins specified under the plugin group are now activated as well. In such cases, it might be better to throw an error.

Reply:
Say my plugin group has the following plugins:
- build:v1.0 (available in staging but deactivated)
- secret:v0.32.0 (available only in production AND activated)

I will not be able to publish this plugin group in activated state using TANZU_CLI_ACTIVATE_PLUGINS_ON_PLUGIN_GROUP_PUBLISH=true in staging because it will not find the secret plugin, which is already activated.

But it may not be worth supporting this case.  For such a  rare case, the plugin developer can just activate the plugins in staging with a separate pipeline instead of using TANZU_CLI_ACTIVATE_PLUGINS_ON_PLUGIN_GROUP_PUBLISH